.github/workflows/prerelease.yml

```diff
@@ -91,158 +91,161 @@ jobs:
-          path: src\repo\exporters\${{ steps.vars.outputs.NAME }}\target\source
       - name: Build windows binary
         id: package
         shell: bash
         run : |
-          make build-${{ steps.vars.outputs.NAME }}
-      - name : Create MSI installer
-        shell: pwsh
-        run: |
-          .\scripts\win_msi_build.ps1 -arch ${{ matrix.goarch }} -exporterName ${{ steps.vars.outputs.NAME }} -version ${{ steps.vars.outputs.VERSION }} -exporterGUID ${{ steps.vars.outputs.EXPORTER_GUID }} -upgradeGUID ${{ steps.vars.outputs.UPGRADE_GUID }} -licenseGUID ${{ steps.vars.outputs.LICENSE_GUID }} -configGUID ${{ steps.vars.outputs.CONFIG_GUID }} -pfx_passphrase "$env:OHAI_PFX_PASSPHRASE"
+          make package-${{ steps.vars.outputs.NAME }}
+
       - name: Test win packages installation
         uses: newrelic/integrations-pkg-test-action/windows@v1
         with:
           tag: ${{ steps.vars.outputs.VERSION  }}
           integration: nri-${{ steps.vars.outputs.NAME }}
-          arch: ${{ matrix.goarch }}
+          arch: amd64
           pkgDir: ${{ github.workspace }}\src\repo\exporters\${{ steps.vars.outputs.NAME }}\target\packages
-          pkgName: ${{ steps.vars.outputs.PACKAGE_NAME }}-${{ matrix.goarch }}.${{ steps.vars.outputs.VERSION }}.msi
+          pkgName: ${{ steps.vars.outputs.PACKAGE_NAME }}-amd64.${{ steps.vars.outputs.VERSION }}.msi
           upgrade: false
-      - name: Upload Release Asset MSI
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload assets to be released
+        uses: actions/upload-artifact@v3
+        with:
+          name: windows-artifacts
+          retention-days: 1
+          path: |
+            ${{ github.workspace }}\src\repo\exporters\${{ steps.vars.outputs.NAME }}\target\packages\*.msi
+            ${{ github.workspace }}\src\repo\exporters\${{ steps.vars.outputs.NAME }}\target\packages\*.zip
+
+  create_publish_schema:
+    name: Create the publish schema
+    runs-on: ubuntu-latest
+    if: ${{ needs.check_exporter_preconditions.outputs.CREATE_RELEASE == 'true' }}
+    needs: [ check_exporter_preconditions ]
+    steps:
+      - uses: actions/checkout@v3
+      - run: make create-publish-schema-${{ needs.check_exporter_preconditions.outputs.INTEGRATION_NAME }}
+      - uses: actions/upload-artifact@v3
         with:
-          upload_url: ${{ needs.create_prerelease.outputs.UPLOAD_URL }}
-          asset_path: ${{ github.workspace }}\src\repo\exporters\${{ steps.vars.outputs.NAME }}\target\packages\${{ steps.vars.outputs.PACKAGE_NAME }}-${{ matrix.goarch }}.${{ steps.vars.outputs.VERSION }}.msi
-          asset_name: ${{ steps.vars.outputs.PACKAGE_NAME }}-${{ matrix.goarch }}.${{ steps.vars.outputs.VERSION }}.msi
-          asset_content_type: msi package
+          name: publish-schema
+          retention-days: 1
+          path: |
+            scripts/pkg/s3-publish-schema-tmp.yml
```

Comment:
why creating the publish schema in a separate job, uploading it and then downloading it?
isn't easier just to call:
```
make create-publish-schema-${{ needs.check_exporter_preconditions.outputs.INTEGRATION_NAME }}
```
?

Reply:
## Another long story...

The publish action did not support `local-custom` when I worked in the pipeline. In fact, I had to create the feature: https://github.com/newrelic/infrastructure-publish-action/pull/142

The first iteration was to upload the artifact using the action and use the API to get the URL of the artifact and use it with the publish action. A nasty quick and dirty hack but that should do the trick.

I learned that we cannot do that. The artifact that you upload are not immediately accessible via API. If you list the artifacts [before](https://github.com/newrelic-forks/newrelic-prometheus-exporters-packages/actions/runs/3794403793/jobs/6452469684#step:6:1) and [after](https://github.com/newrelic-forks/newrelic-prometheus-exporters-packages/actions/runs/3794403793/jobs/6452469684#step:8:1) uploading and artifact using [the action](https://github.com/newrelic-forks/newrelic-prometheus-exporters-packages/actions/runs/3794403793/jobs/6452469684#step:7:1) there are no changes. Artifacts are only available AFTER the workflow has finished.

[Link to the docs](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#downloading-or-deleting-artifacts) but I'll save you a click: 
> ### Downloading or deleting artifacts
> During a workflow run, you can use the download-artifact action to download artifacts that were previously uploaded in the same workflow run.
> After a workflow run has been completed, you can download or delete artifacts on GitHub or using the REST API.

Note that "**After a workflow run has been completed**".

How does the action to download fetch the artifacts? After reverse engineer [the action calls a URL](https://github.com/actions/download-artifact/blob/9782bd6/dist/index.js#L1769) like `${config_variables_1.getRuntimeUrl()}_apis/pipelines/workflows/${config_variables_1.getWorkFlowRunId()}/artifacts?api-version=${getApiVersion()}`. The function `getRuntimeUrl()` [simply looks](https://github.com/actions/download-artifact/blob/9782bd6a9848b53b110e712e20e42d89988822b7/dist/index.js#L255-L261) for the environment variable `ACTIONS_RUNTIME_URL`.

The environment variable `ACTIONS_RUNTIME_URL` is only available INSIDE an action but not in a step that runs a script. If I run a step with `set` that variable does not exist. These internal environment variables are undocumented: https://github.com/nektos/act/issues/329

Someone created an action that dumps the environment variables from inside an action and you can see that environment variable there: https://gist.github.com/sonots/070f1059b71dd2dec8e43ec513c3f8f0

Then you'll ask, why did it survive after this failure?

It was really useful to have it uploaded for debugging purposes as you can fetch it and see how it templates just in case the `yq` that you have locally is not the same `yq` that GitHub is using (yes, there are two `yq` called `yq` in Python and Go).

It parallelizes (a bit) the jobs and removes the need to check out the code (again) in the last job when the artifacts are uploaded making that job half a minute shorter.